### ISSUE TITLE
Block placement flickering while trying to place block too close to yourself

### ISSUE THREAD

**Issue by ItsZodiaX** [pmmp/PocketMine-MP#4924]
### Issue description

- Expected result: Block placement does not flickering
- Actual result: Block placement does flickering

### Steps to reproduce the issue
1. Trying to place block under your feet or in front of your face

### OS and versions
* PocketMine-MP: 4.2.4
* PHP: 8.0.17
* Using JIT: No
* Server OS: Linux (Ubuntu 20.04)
* Game version: Android/iOS/Win10/Xbox/PS4/Switch

### Plugins
Plugins (0):

https://cdn.discordapp.com/attachments/878545340058468412/956586777471234108/2565-03-24_23.13.58.png

- If you remove all plugins, does the issue still occur? Yes

### Crashdump, backtrace or other files
PocketMine-MP : 
https://cdn.discordapp.com/attachments/878545340058468412/956585496505626714/2022-03-24_22-56-34.mp4
https://cdn.discordapp.com/attachments/878545340058468412/955953635450302474/Minecraft_2022.03.23_-_05.44.15.08.DVR_Trim.mp4

Alternative server software (Dragonfly) : https://cdn.discordapp.com/attachments/878545340058468412/956585775984705556/2022-03-24_23-04-10.mp4

**Comment by dktapps** [comment]
I've been aware of this issue for a while.

Other servers have worked around this issue by avoiding resends if the collision boxes of the placed block overlap with that of the player. However, this is not a reliable solution because it assumes that PM has perfect vanilla knowledge. Any mismatch would result in ghost blocks instead of flickering, which is arguably worse.

Any other potential solution would require some kind of throttling of the right-click spam to avoid the spammy block resends that cause this problem.

**Comment by ItsZodiaX** [comment]
Does holding count as spamming? Because i was holding the right click in the video.

**Comment by dktapps** [comment]
Yes. The client continuously sends "right click" interactions while you're holding down the right clicker, regardless of whether something would actually result from doing so.

**Comment by ItsZodiaX** [comment]
So what we should do is a rework on hacky method that prevent right click spamming?

**Comment by dktapps** [comment]
Not exactly. If the hack that PM currently uses was removed, the problem would be worsened.

**Comment by ItsZodiaX** [comment]
So this issue can not be fixed with code?

**Comment by dktapps** [comment]
A perfect solution for this is not possible with current protocol, only workarounds.

**Comment by ItsZodiaX** [comment]
Thanks for the info. Hopefully this issue can be fixed in this nearly future. You can close this issue.
